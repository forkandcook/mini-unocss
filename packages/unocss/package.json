{
  "name": "unocss",
  "version": "0.1.0-beta.2",
  "description": "",
  "keywords": [],
  "homepage": "https://github.com/antfu/unocss#readme",
  "bugs": {
    "url": "https://github.com/antfu/unocss/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/antfu/unocss.git"
  },
  "funding": "https://github.com/sponsors/antfu",
  "license": "MIT",
  "author": "Anthony Fu <anthonyfu117@hotmail.com>",
  "sideEffects": false,
  "exports": {
    ".": {
      "require": "./dist/index.js",
      "import": "./dist/index.mjs"
    },
    "./vite": {
      "require": "./dist/vite.js",
      "import": "./dist/vite.mjs"
    }
  },
  "main": "dist/index.js",
  "module": "dist/index.mjs",
  "types": "index.d.ts",
  "files": [
    "dist",
    "*.d.ts"
  ],
  "scripts": {
    "build": "tsup",
    "dev": "tsup --watch src",
    "prepublishOnly": "nr build"
  },
  "dependencies": {
    "@unocss/core": "workspace:*",
    "@unocss/preset-attributify": "workspace:*",
    "@unocss/preset-uno": "workspace:*",
    "@unocss/vite": "workspace:*"
  },
  "devDependencies": {
    "nanoevents": "^6.0.2"
  }
}
