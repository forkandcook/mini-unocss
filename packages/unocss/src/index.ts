export * from '@unocss/core'
export { default as presetUno } from '@unocss/preset-uno'
export { default as presetAttributify } from '@unocss/preset-attributify'
