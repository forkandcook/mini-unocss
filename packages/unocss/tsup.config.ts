import { Options } from 'tsup'

const config: Options = {
  splitting: true,
  format: ['esm', 'cjs'],
  entryPoints: [
    'src/*.ts',
  ],
  clean: true,
  dts: true,
}

export default config
