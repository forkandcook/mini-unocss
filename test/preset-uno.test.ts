import { createGenerator, escapeSelector, presetUno } from 'unocss'

const classes = [
  '-gap-y-5',
  '-m-auto',
  '!hover:px-10',
  '!p-5px',
  'all:m-auto',
  'bg-[#153]/10',
  'bg-[#1533]',
  'bg-[#1533]/10',
  'bg-#452233/40',
  'bg-hex-452233/40',
  'bg-opacity-45',
  'bg-red-100',
  'bg-teal-100/55',
  'blur-4',
  'border-2',
  'border-b',
  'border-green-100/10',
  'border-t-2',
  'border',
  'children:m-auto',
  'dark:not-odd:text-red',
  'dark:text-xl',
  'duration-111',
  'flex-[hi]',
  'flex',
  'font-mono',
  'gap-4',
  'gap-x-1',
  'grid-cols-[1fr,2fr,100px,min-content]',
  'grid-cols-2',
  'grid-rows-[1fr,2fr,100px,min-content]',
  'grid-rows-3',
  'grid',
  'h-1',
  'hover:!p-10',
  'hover:not-first:checked:bg-red/10',
  'hover:p-4',
  'leading-2',
  'light:text-sm',
  'm-[3em]',
  'm-0',
  'm-1/2',
  'm-auto',
  'max-h-[1px]',
  'md:!hidden',
  'md:m-1',
  'my-auto',
  'op-10',
  'opacity-0',
  'order-first',
  'overflow-auto',
  'overflow-x-scroll',
  'p-2',
  'p-t-2',
  'p2',
  'pl-10px',
  'pt-2',
  'pt2',
  'rounded-[4px]',
  'rounded-1/2',
  'rounded-full',
  'rounded-md',
  'rounded-rb-1/2',
  'rounded-t-sm',
  'rounded-tr',
  'rounded',
  'sm:m-1',
  'sm:m1',
  'lt-sm:m1',
  'lt-lg:m2',
  'text-[#124]',
  'text-4xl',
  'text-base',
  'text-black/10',
  'text-blue',
  'text-lg',
  'text-red-100',
  'text-red-200/10',
  'text-red-300/20',
  'text-red100',
  'text-red2',
  'top-0',
  'tracking-wide',
  'transition-200',
  'transition',
  'w-1/2',
  'z-1',
  'z-100',
]

const code = classes.join(' ')
const uno = createGenerator({
  presets: [
    presetUno(),
  ],
})

test('default', async() => {
  const { css } = await uno.generate(code)
  const { css: css2 } = await uno.generate(code)

  const unmatched = []
  for (const i of classes) {
    if (!css.includes(escapeSelector(i)))
      unmatched.push(i)
  }
  expect(unmatched).toEqual([])
  expect(css).toMatchSnapshot()
  expect(css).toEqual(css2)
})
