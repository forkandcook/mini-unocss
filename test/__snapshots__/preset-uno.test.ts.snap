// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default 1`] = `
".\\\\!p-5px{padding:5px !important;}
.hover\\\\:\\\\!p-10:hover{padding:2.5rem !important;}
.hover\\\\:p-4:hover{padding:1rem;}
.p-2,.p2{padding:0.5rem;}
.\\\\!hover\\\\:px-10:hover{padding-left:2.5rem !important;padding-right:2.5rem !important;}
.pl-10px{padding-left:10px;}
.pt-2,.p-t-2,.pt2{padding-top:0.5rem;}
.m-\\\\[3em\\\\]{margin:3em;}
.m-0{margin:0rem;}
.m-1\\\\/2{margin:50%;}
.children\\\\:m-auto > *,.all\\\\:m-auto *,.-m-auto,.m-auto{margin:auto;}
.my-auto{margin-top:auto;margin-bottom:auto;}
.op-10{opacity:0.1;}
.opacity-0{opacity:0;}
.bg-\\\\[\\\\#1533\\\\]{background-color:rgba(17,85,51,0.2);}
.bg-\\\\[\\\\#153\\\\]\\\\/10,.bg-\\\\[\\\\#1533\\\\]\\\\/10{background-color:rgba(17,85,51,0.1);}
.bg-\\\\#452233\\\\/40,.bg-hex-452233\\\\/40{background-color:rgba(69,34,51,0.4);}
.bg-red-100{--un-bg-opacity:1;background-color:rgba(254,226,226,var(--un-bg-opacity));}
.bg-teal-100\\\\/55{background-color:rgba(204,251,241,0.55);}
.hover\\\\:not-first\\\\:checked\\\\:bg-red\\\\/10:hover:not(:first):checked{background-color:rgba(248,113,113,0.1);}
.bg-opacity-45{--un-bg-opacity:0.45;}
.border{border-width:1px;border-style:solid;}
.border-2{border-width:2px;border-style:solid;}
.border-b{border-bottom-width:1px;border-style:solid;}
.border-t-2{border-top-width:2px;border-style:solid;}
.border-green-100\\\\/10{border-color:rgba(220,252,231,0.1);}
.rounded{border-radius:0.25rem;}
.rounded-\\\\[4px\\\\]{border-radius:4px;}
.rounded-1\\\\/2{border-radius:50%;}
.rounded-full{border-radius:9999px;}
.rounded-md{border-radius:0.375rem;}
.rounded-rb-1\\\\/2{border-bottom-right-radius:50%;}
.rounded-t-sm{border-top-left-radius:0.125rem;border-top-right-radius:0.125rem;}
.rounded-tr{border-top-right-radius:0.25rem;}
.font-mono{font-family:ui-monospace,SFMono-Regular,Menlo,Monaco,Consolas,\\"Liberation Mono\\",\\"Courier New\\",monospace;}
.dark .dark\\\\:text-xl{font-size:1.25rem;line-height:1.75rem;}
.light .light\\\\:text-sm{font-size:0.875rem;line-height:1.25rem;}
.text-4xl{font-size:2.25rem;line-height:2.5rem;}
.text-base{font-size:1rem;line-height:1.5rem;}
.text-lg{font-size:1.125rem;line-height:1.75rem;}
.dark .dark\\\\:not-odd\\\\:text-red:not(:nth-child(odd)){--un-text-opacity:1;color:rgba(248,113,113,var(--un-text-opacity));}
.text-\\\\[\\\\#124\\\\]{--un-text-opacity:1;color:rgba(17,34,68,var(--un-text-opacity));}
.text-black\\\\/10{color:rgba(0,0,0,0.1);}
.text-blue{--un-text-opacity:1;color:rgba(96,165,250,var(--un-text-opacity));}
.text-red-200\\\\/10{color:rgba(254,202,202,0.1);}
.text-red-300\\\\/20{color:rgba(252,165,165,0.2);}
.text-red-100,.text-red100{--un-text-opacity:1;color:rgba(254,226,226,var(--un-text-opacity));}
.text-red2{--un-text-opacity:1;color:rgba(254,202,202,var(--un-text-opacity));}
.flex-\\\\[hi\\\\]{flex:hi;}
.flex{display:flex;}
.grid{display:grid;}
.grid-cols-2{grid-template-columns:repeat(2, minmax(0, 1fr));}
.grid-rows-3{grid-template-rows:repeat(3, minmax(0, 1fr));}
.grid-cols-\\\\[1fr\\\\2c 2fr\\\\2c 100px\\\\2c min-content\\\\]{grid-template-columns:1fr 2fr 100px min-content;}
.grid-rows-\\\\[1fr\\\\2c 2fr\\\\2c 100px\\\\2c min-content\\\\]{grid-template-rows:1fr 2fr 100px min-content;}
.gap-4{grid-gap:1rem;gap:1rem;}
.gap-x-1{grid-column-gap:0.25rem;column-gap:0.25rem;}
.-gap-y-5{grid-row-gap:-1.25rem;row-gap:-1.25rem;}
.w-1\\\\/2{width:50%;}
.h-1{height:0.25rem;}
.max-h-\\\\[1px\\\\]{max-height:1px;}
.tracking-wide{letter-spacing:0.025em;}
.leading-2{line-height:0.5rem;}
.overflow-auto{overflow:auto;}
.overflow-x-scroll{overflow-x:scroll;}
.order-first{order:-9999;}
.top-0{top:0rem;}
.z-1{z-index:1;}
.z-100{z-index:100;}
.transition{transition-property:background-color, border-color, color, fill, stroke, opacity, box-shadow, transform, filter, backdrop-filter;transition-timing-function:cubic-bezier(0.4, 0, 0.2, 1);transition-duration:150ms;}
.transition-200{transition-property:background-color, border-color, color, fill, stroke, opacity, box-shadow, transform, filter, backdrop-filter;transition-timing-function:cubic-bezier(0.4, 0, 0.2, 1);transition-duration:200ms;}
.duration-111{transition-duration:111ms;}
.blur-4{--un-blur:blur(4px);filter:var(--un-blur) var(--un-brightness) var(--un-contrast) var(--un-grayscale) var(--un-hue-rotate) var(--un-invert) var(--un-saturate) var(--un-sepia) var(--un-drop-shadow);}
@media (min-width: 768px){
.md\\\\:m-1{margin:0.25rem;}
.md\\\\:\\\\!hidden{display:none !important;}
}
@media (min-width: 640px){
.sm\\\\:m-1,.sm\\\\:m1{margin:0.25rem;}
}
@media (max-width: 640px){
.lt-sm\\\\:m1{margin:0.25rem;}
}
@media (max-width: 1024px){
.lt-lg\\\\:m2{margin:0.5rem;}
}"
`;
